hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/RecordStore.java

```diff
@@ -465,11 +465,12 @@ void doPostEvictionOperations(@Nonnull Data dataKey, @Nonnull Object dataValue,
     /**
      * Returns live record or null if record is already expired. Does not load missing keys from a map store.
      *
-     * @param key key to be accessed
+     * @param key      key to be accessed
+     * @param backup true if partition is a backup-partition otherwise set false
      * @return live record or null
      * @see #get
      */
-    R getRecordOrNull(Data key);
+    R getRecordOrNull(Data key, boolean backup);
```

Comment:
Can you not replace existing method but add this as a new API? You can make existing into a default method. Also in javadoc, you can explain why it's needed (e.g. lock related reasons) and maybe also point out to newly added tests.

Reply:
I want to remove it because it is easy to misuse, method must always be called with backup flag. 
Since this is not specific to lock related issue, current javaDoc should be sufficient. Lock related issue is where it made itself visible and this method is used as a part of different use cases as well.